### PR TITLE
Handle start before DHCP resolution

### LINKED ISSUE #432: Handle start before DHCP resolution
in some logs we have seen the following failure in our MDNS advertisement:

```
Dec 22 01:06:22 amplipi authbind[807]: AmpliPi zeroconf - unable to register service on one of ['eth0', 'wlan0', 'eth0'],             they all are either not available or have no IP address.
Dec 22 01:06:22 amplipi authbind[807]: AmpliPi zeroconf - is this running on AmpliPi?
Dec 22 01:06:22 amplipi authbind[807]: AmpliPi zeroconf - registering service: ServiceInfo(type='_amplipi._tcp.local.', name='amplipi-none._amplipi._tcp.local.', addresses=[b'\x00\x00\x00\x00'], port=80, weight=0, priority=0, server='amplipi.local.', properties={'path': '/api/', 'name': 'AmpliPi', 'vendor': 'MicroNova', 'version': '0.1.9', 'web_app': 'http://amplipi.local/', 'documentation': 'http://amplipi.local/doc'}, interface_index=None)
Dec 22 01:06:22 amplipi authbind[807]: Process Process-1:
Dec 22 01:06:22 amplipi authbind[807]: Traceback (most recent call last):
Dec 22 01:06:22 amplipi authbind[807]:   File "/usr/lib/python3.7/multiprocessing/process.py", line 297, in _bootstrap
Dec 22 01:06:22 amplipi authbind[807]:     self.run()
Dec 22 01:06:22 amplipi authbind[807]:   File "/usr/lib/python3.7/multiprocessing/process.py", line 99, in run
Dec 22 01:06:22 amplipi authbind[807]:     self._target(*self._args, **self._kwargs)
Dec 22 01:06:22 amplipi authbind[807]:   File "./amplipi/app.py", line 870, in advertise_service
Dec 22 01:06:22 amplipi authbind[807]:     zeroconf = Zeroconf(ip_version=IPVersion.V4Only, interfaces=[ip_addr])
Dec 22 01:06:22 amplipi authbind[807]:   File "/home/pi/amplipi-dev/venv/lib/python3.7/site-packages/zeroconf/_core.py", line 410, in __init__
Dec 22 01:06:22 amplipi authbind[807]:     listen_socket, respond_sockets = create_sockets(interfaces, unicast, ip_version, apple_p2p=apple_p2p)
Dec 22 01:06:22 amplipi authbind[807]:   File "/home/pi/amplipi-dev/venv/lib/python3.7/site-packages/zeroconf/_utils/net.py", line 359, in create_sockets
Dec 22 01:06:22 amplipi authbind[807]:     if add_multicast_member(cast(socket.socket, listen_socket), i):
Dec 22 01:06:22 amplipi authbind[807]:   File "/home/pi/amplipi-dev/venv/lib/python3.7/site-packages/zeroconf/_utils/net.py", line 276, in add_multicast_member
Dec 22 01:06:22 amplipi authbind[807]:     listen_socket.setsockopt(socket.IPPROTO_IP, socket.IP_ADD_MEMBERSHIP, _value)
```

It appears that this code may be running before DHCP has resolved an ip address.

A couple options to handler this:
1. Make the AmpliPi service use the dependency After=`network-online.target` instead of `network.target` (this can take up to 90 seconds in a no network available situation)
2. Retry several times when an ip address is not available to broadcast on
3. Continue retrying until a network interface with an ip is available

### PR DESCRIPTION
**Pull request by linknum23**
Handle start before DHCP resolution
closes #432 

Also try and include #579

### CODE REVIEW

**Comment by linknum23** [comment]
Added support for DHCP delay and IP changed mid flight, needs testing!

File: amplipi/app.py [hunk 1]
```diff
@@ -807,6 +807,24 @@ def advertise_service(port, que: Queue):
...
   """
+  def ok():
+    """ Was a stop requested by the parent process? """
+    return que.empty()
+
+  while ok():
+    try:
+      _advertise_service(port, ok)
+    except Exception as e:
+      print(f'Failed to advertise AmpliPi service: {e}')
```
Comment (linknum23) [inline amplipi/app.py:818]: TODO: better reporting (An IP change is not really a failure)

**Comment by rtertiaer** [comment]
@linknum23  if you intend to ship this eventually, can you re-target this PR to `main`? Github automagically closed it when I deleted `develop` :woozy_face: sorry!

**Comment by linknum23** [comment]
This fixes a known bug. It just needs a little TLC.

**Pull request closed without merging**